mimesis/locales.py

```diff
@@ -38,30 +38,162 @@
+    },
+    EN: {
+        'name': 'English',
+        'name_local': 'English',
+    },
+    EN_GB: {
+        'name': 'British English',
+        'name_local': 'English',
+    },
+    EN_AU: {
+        'name': 'Australian English',
+        'name_local': 'English',
+    },
+    EN_CA: {
+        'name': 'Canadian English',
+        'name_local': 'English',
+    },
+    ES: {
+        'name': 'Spanish',
+        'name_local': 'Español',
+    },
+    ES_MX: {
+        'name': 'Mexican Spanish',
+        'name_local': 'Español',
+    },
+    ET: {
+        'name': 'Estonian',
+        'name_local': 'Eesti',
+    },
+    FA: {
+        'name': 'Farsi',
+        'name_local': 'فارسی',
+    },
+    FI: {
+        'name': 'Finnish',
+        'name_local': 'Suomi',
+    },
+    FR: {
+        'name': 'French',
+        'name_local': 'Français',
+    },
+    HU: {
+        'name': 'Hungarian',
+        'name_local': 'Magyar',
+    },
+    IS: {
+        'name': 'Icelandic',
+        'name_local': 'Íslenska',
+    },
+    IT: {
+        'name': 'Italian',
+        'name_local': 'Italiano',
+    },
+    JA: {
+        'name': 'Japanese',
+        'name_local': '日本語',
+    },
+    KK: {
+        'name': 'Kazakh',
+        'name_local': 'Қазақша',
+    },
+    KO: {
+        'name': 'Korean',
+        'name_local': '한국어',
+    },
+    NL: {
+        'name': 'Dutch',
+        'name_local': 'Nederlands',
+    },
+    NL_BE: {
+        'name': 'Belgium Dutch',
+        'name_local': 'Nederlands',
+    },
+    NO: {
+        'name': 'Norwegian',
+        'name_local': 'Norsk',
+    },
+    PL: {
+        'name': 'Polish',
+        'name_local': 'Polski',
+    },
+    PT: {
+        'name': 'Portuguese',
+        'name_local': 'Português',
+    },
+    PT_BR: {
+        'name': 'Brazilian Portuguese',
+        'name_local': 'Português Brasileiro',
+    },
+    RU: {
+        'name': 'Russian',
+        'name_local': 'Русский',
+    },
+    SV: {
+        'name': 'Swedish',
+        'name_local': 'Svenska',
+    },
+    TR: {
+        'name': 'Turkish',
+        'name_local': 'Türkçe',
+    },
+    UK: {
+        'name': 'Ukrainian',
+        'name_local': 'Українська',
+    },
+    ZH: {
+        'name': 'Chinese',
+        'name_local': '汉语',
+    },
+}
+
+LIST_OF_LOCALES = list(SUPPORTED_LOCALES)
 
 
 @contextlib.contextmanager
-def override(provider: DataProviderType,
-             locale: str = EN) -> Generator[DataProviderType, None, None]:
+def override(provider: 'BaseDataProvider',
+             locale: str = EN) -> Generator['BaseDataProvider', None, None]:
     """Context manager which allows overriding current locale.
 
     :param provider: Locale dependent data provider.
     :param locale: Locale.
     :return:
     """
     origin_locale = getattr(provider, 'locale', None)
-    datafile = getattr(provider, '_datafile', None)
-
-    if not datafile or not locale:
+    if not hasattr(provider, '_datafile') or not origin_locale:
```

Comment:
I don't like this solution. It is better to ask for forgiveness that to ask for permission.
I would change it to either:
1. raise an exception when attribute does not exist 
2. check base class, we have a special base class for locales now, why not use it?

You decide which one you like more.